tests/segfault.phpt

```diff
@@ -0,0 +1,23 @@
+--TEST--
+Segfault when re-taint a string which lost its taint flag
+--SKIPIF--
+<?php if (!extension_loaded("taint")) print "skip"; ?>
+--INI--
+taint.enable=1
+--FILE--
+<?php 
+$a = "tainted string" . ".";
+taint($a); //must use concat to make the string not a internal string(introduced in 5.4)
+var_dump(is_tainted($a));
+
+$a = preg_replace('{^\s*SELECT}i', 'SELECT/*', $a, 1);
+var_dump(is_tainted($a));
```

Comment:
atm this is FALSE which is a bug in my eyes per https://github.com/laruence/taint/pull/49